hera_cal/lstbin_simple.py

```diff
@@ -840,13 +974,31 @@ def lst_bin_files_single_outfile(
     flag_thresh: float = 0.7,
     freq_min: float | None = None,
     freq_max: float | None = None,
+    output_inpainted: bool | None = None,
+    output_flagged: bool  = True,
+    where_inpainted_file_rules: list[tuple[str, str]] | None = None,
 ) -> dict[str, Path]:
     """
     Bin data files into LST bins, and write all bins to disk in a single file.
 
     Note that this is generally not meant to be called directly, but rather through
     the :func:`lst_bin_files` function.
 
+    The mode(s) in which the function does the averaging can be specified via the
+    `output_inpainted`, `output_flagged` and `where_inpainted_file_rules` options.
+    Algorithmically, there are two modes of averaging: either flagged data is *ignored*
+    in the average (and in Nsamples) or the flagged data is included in the average but
+    ignored in Nsamples. These are called "flagged" and "inpainted" modes respectively.
+    The latter only makes sense if the data in the flagged regions has been inpainted,
+    rather than left as raw data. For delay-filtered data, either mode is equivalent,
+    since the flagged data itself is set to zero. By default, this function *only*
+    uses the "flagged" mode, *unless* the `where_inpainted_file_rules` option is set,
+    which indicates that the files are definitely in-painted, and in this case it will
+    use *both* modes by default.
```

Comment:
I'm not sure what the most obvious choice is. Both can be computed in the same run for MUCH cheaper than doing two separate runs (avoiding re-reading the data), so I thought it would be good to "highlight" that fact by making it the default if possible. Of course, either mode can be turned off/on explicitly as well.

Reply:
Got it. Fair enough.